### DISCUSSION TITLE
Where is (auth) Display Name populated from?

### DISCUSSION THREAD

**Discussion by jaredakers** [supabase/supabase#20060] [Questions] [answered]
Using auth.SignUp with email, password, in some testing i ended up with an first and last name in the Display Name field of the auth table. However, for the life of me I can't recall how I did that. And can't replicate it. 

I am using a public profile table as well, but would like the display name in the auth table as well. 

Any idea how to populate the Display Name field? 

![image](https://github.com/supabase/supabase/assets/4602606/376cdc56-6a34-4176-82be-a623221d9bd8)

**Comment by GaryAustin1** [comment, marked as answer]
This comes from OAuth providers and is stored in the raw_user_meta_data column.  It is derived from many possible keys there but display_name is one of them.

**Reply by jaredakers** [reply]
thanks!

**Reply by MarkOSullivan94** [reply]
Do you have a list of all the possible keys it can be derived from?

**Reply by GaryAustin1** [reply]
I just searched Supabase's source files at supabase/supabase to find displayname.
https://github.com/supabase/supabase/blob/df52ea7ee0f0c76ba952ac09c6182fe79824eef0/apps/studio/components/interfaces/Auth/Users/UserListItem.utils.ts#L6

**Comment by robpetty** [comment]
Haven't seen the code but wanted to add, I am not using display_name and stumbled into this simply by adding first and last name collection. In my public.users (terrible I know should have public.profile), I added a display_name that was independent of anything, so probably will rework it now to just be first + last (got a trigger/func to move the names into the public.users).

```
const { error } = await supabase.auth.signUp({
    email,
    password,
    options: {
      data: {
        first_name: firstName,
        last_name: lastName,
      },
      emailRedirectTo: `${requestUrl.origin}/auth/callback`,
    },
  });
```

**Reply by bryanprimus** [reply]
using `first_name` and `last_name` working?

**Reply by daedcpl** [reply]
Adding first & last names in data worked for me.

**Reply by GiannIsidore** [reply]
how to access that?

**Reply by lklyne** [reply]
Looks like the display name can be accessed like this
```
const { data, error } = await supabase.auth.getUser()
const displayName = data.user.user_metadata.display_name
```

**Reply by BugBuster-404** [reply]
> Looks like the display name can be accessed like this
> 
> ```
> const { data, error } = await supabase.auth.getUser()
> const displayName = data.user.user_metadata.display_name
> ```

This worked for me, Thank mate!

**Reply by rabin-thami** [reply]
What if I want to update a field in that form, like the full name?

**Reply by battybrats** [reply]
any updates on this? updating full name

edit: found the solution by change

`
let { data, error } = await supabase.auth.updateUser({
      data: { full_name },
    });
`

**Reply by caseyocampo** [reply]
For anyone wondering how to update display_name, here's what my action looks like:

```
const data = {
    display_name: formData.get("display_name") as string,
  };

  const { error } = await supabase.auth.updateUser({ data });
```

**Comment by pootytang** [comment]
Wanted to add this here just in case someone else comes across this.  I'm using sveltekit and in my +page.server.ts action, I'm doing this:
```javascript
  const creds: SignUpWithPasswordCredentials = {
      email,
      password,
      options: {
        data: {display_name: `${fName} ${lName}`}
      }
    }
    const {data, error} = await supabase.auth.signUp(creds)
```

**Comment by hecanet** [comment]
Print the enter metadata and pick what you want from it.

print(user?.userMetadata);

In my case, I wanted to full name. So, here is what I implemented:

user?.userMetadata?['full_name'] ?? "Full Name",

**Comment by AndeYashwanth** [comment]
Is this the right way to get full name in kotlin? I dont see many examples on userMetadata usage in the documentation.
val fullName = (authenticatedUser.userMetadata?.get("full_name") as? JsonPrimitive)?.content ?: "Unknown"

**Reply by GaryAustin1** [reply]
The user metadata will contain the naming provided by the signUP call or the OAuth provider I'm pretty sure.
The post I made above shows how the dashboard panel takes that and builds a name from the many different names that could be in the metadata.
